pig-runtime/src/org/partiql/pig/runtime/DomainVisitorFoldBase.kt

```diff
@@ -38,6 +42,12 @@ open class DomainVisitorFoldBase<T> {
 
     ///////////////////////////////////////////////////////
 
+    open fun walkBoolPrimitive(node: BoolPrimitive, accumulator: T): T {
+        var current = accumulator
+        current = visitBoolPrimitive(node, current)
+        return walkMetas(node.metas, current)
```

Comment:
```suggestion
        val current = visitBoolPrimitive(node, accumulator)
        return walkMetas(node.metas, current)
```
And similarly for other `walkPrimitive` APIs.

Reply:
sure... this was following the pattern of the generated code which does it that way to simplify the template... but yes, this could be simplified in hand-written code.